portal-impl/src/com/liferay/portlet/internal/ClientDataRequestImpl.java

```diff
@@ -172,6 +172,15 @@ public void setCharacterEncoding(String enc)
 		for (Map.Entry<String, String[]> entry : parameterMap.entrySet()) {
 			String name = entry.getKey();
 
+			// If the parameter name is not a public/private render parameter,
```

Comment:
@shuyangzhou These are some comments that Brian permitted in a previous merge that got removed when you moved common functionality to ClientDataRequestImpl in a code review commit. Please consider letting them stay as they will be helpful to me in future maintenance.